File: try-asyncio.js
const sleep = (seconds) => {
	return new Promise((resolve, reject) => {
		setTimeout(() => {
			resolve();
		}, seconds * 1000);
	});
};

async function count() {
	console.log("one");
	await sleep(1);
	console.log("two");
}

async function main() {
	const startTime = new Date().getTime();
	
	console.time("main");

	await count();
	await count();
	await count();

	console.timeEnd("main");
	
	const endTime = new Date().getTime();
	console.log(`elapsed time: ${endTime - startTime}`)
}

main();

File: try-asyncio.py
import asyncio

async def count():
	print("one")
	await asyncio.sleep(1)
	print("two")

async def main():
	await asyncio.gather(count(), count(), count())

if __name__ == "__main__":
	import time
	start_time = time.perf_counter()

	asyncio.run(main())

	elapsed_time = time.perf_counter() - start_time
	print(f"{__file__} executed in {elapsed_time:0.2f} seconds")
